apps/studio/components/interfaces/SQLEditor/UtilityPanel/UtilityActions.tsx

```diff
@@ -46,48 +63,70 @@ const UtilityActions = ({
   )
 
   return (
-    <div className="inline-flex items-center justify-end gap-x-2">
-      {IS_PLATFORM && <SavingIndicator id={id} />}
+    <div className="inline-flex items-center justify-end gap-x-4">
+      <div className="flex items-center justify-center gap-x-2">
+        {IS_PLATFORM && <SavingIndicator id={id} />}
+
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              type="text"
+              className="px-1"
+              icon={<Keyboard size={14} className="text-foreground-light" />}
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="w-48">
+            <DropdownMenuItem
+              className="justify-between"
+              onClick={() => {
+                setIntellisenseEnabled(!intellisenseEnabled)
+                toast.success(
+                  `Successfully ${intellisenseEnabled ? 'disabled' : 'enabled'} intellisense. ${intellisenseEnabled ? 'Please refresh your browser for changes to take place.' : ''}`
+                )
+              }}
+            >
+              <p>Intellisense enabled</p>
+              {intellisenseEnabled && <Check className="text-brand" size={16} />}
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+
+        {IS_PLATFORM && <FavoriteButton id={id} />}
+
+        <Tooltip_Shadcn_>
+          <TooltipTrigger_Shadcn_ asChild>
+            <Button
+              type="text"
+              onClick={() => prettifyQuery()}
+              className="px-1"
+              icon={<AlignLeft size="tiny" strokeWidth={2} />}
+            />
+          </TooltipTrigger_Shadcn_>
+          <TooltipContent_Shadcn_ side="bottom">Prettify SQL</TooltipContent_Shadcn_>
+        </Tooltip_Shadcn_>
+      </div>
 
       <DropdownMenu>
         <DropdownMenuTrigger asChild>
-          <Button
-            type="text"
-            className="px-1"
-            icon={<Keyboard size={14} className="text-foreground-light" />}
-          />
+          <Button asChild type="outline" iconRight={<ChevronDown size={14} />}>
+            <span>{ROWS_PER_PAGE_OPTIONS.find((opt) => opt.value === snap.limit)?.label}</span>
```

Comment:
span ?

Reply:
its a workaround for the removeChild / insertBefore error that we see when users use Google Translate (or some third party extension) - although i'm genuinely not sure whether this is worth pursuing to keep doing as a practice tbh ([ref](https://github.com/supabase/supabase/pull/26580))

atm inclined to remove the span tbh